src/Libraries/DSCPython/DCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,806 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            return assignments;
+        }
+
+        private Type TryGetType(string name)
+        {
+            if (ImportedTypes.ContainsKey(name))
+            {
+                return ImportedTypes[name];
+            }
+
+            // If the type name does not exist in the local or built-in variables, then it is out of scope
+            string lookupScr = String.Format("clr.GetClrType({0}) if (\"{0}\" in locals() or \"{0}\" in __builtins__) and isinstance({0}, type) else None", name);
+
+            dynamic type = null;
+            try
+            {
+                type = ExecutePythonScriptCode(lookupScr);
+            }
+            catch (Exception e)
+            {
+                Log(String.Format("Failed to look up type: {0}", name));
+                Log(e.ToString());
+            }
+
+            var foundType = type as Type;
+            if (foundType != null)
+            {
+                ImportedTypes[name] = foundType;
+            }
+
+            return foundType;
+        }
+
+        /// <summary>
+        /// List all of the members in a PythonModule
+        /// </summary>
+        /// <param name="module">A reference to the module</param>
+        /// <param name="name">The name of the module</param>
+        /// <returns>A list of completion data for the module</returns>
+        private IEnumerable<Tuple<string, string, bool, ExternalCodeCompletionType>> EnumerateMembers(object module, string name)
+        {
+            var items = new List<Tuple<string, string, bool, ExternalCodeCompletionType>>();
+            var d = (module as PyObject);
+
+            foreach (var member in d.Dir())
+            {
+                try
+                {
+                    var completionType = ExternalCodeCompletionType.Field;
+
+                    if (d.GetAttr(member.ToString()).ToString().Contains(PythonCodeCompletionUtils.inBuiltMethod) ||
+                        d.GetAttr(member.ToString()).ToString().Contains(PythonCodeCompletionUtils.method))
+                    {
+                        completionType = ExternalCodeCompletionType.Method;
+                    }
+
+                    items.Add(Tuple.Create((string)member.ToString(), name, false, completionType));
+                }
+                catch (Exception ex)
+                {
+                    Log(ex.ToString());
+                }
+            }
+
+            return items;
+        }
+
+        /// <summary>
+        /// List all of the members in a CLR type
+        /// </summary>
+        /// <param name="type">The type</param>
+        /// <param name="name">The name for the type</param>
+        /// <returns>A list of completion data for the type</returns>
+        private IEnumerable<Tuple<string, string, bool, ExternalCodeCompletionType>> EnumerateMembers(Type type, string name)
```

Comment:
have you tested that python objects (not clr objects) get auto complete? For example, try creating a list or dictionary directly in Python - does auto complete work?

Reply:
I does not seem to work for python native types. 
Same thing for IronPythonAutoComplete..